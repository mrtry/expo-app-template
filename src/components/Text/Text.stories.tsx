import type { Meta, StoryObj } from '@storybook/react';

import { ScrollView, View } from 'react-native';

import { Text } from './index';

const meta = {
  component: Text,
  tags: ['autodocs'],
  title: 'Components/Text',
} satisfies Meta<typeof Text>;

export default meta;

type Story = StoryObj<typeof meta>;

export const AllVariants: Story = {
  args: {
    children: 'Placeholder',
    variant: 'bodyMedium',
  },
  render: () => (
    <ScrollView style={{ flex: 1 }}>
      <View style={{ gap: 16, padding: 16 }}>
        <View style={{ marginBottom: 8 }}>
          <Text style={{ marginBottom: 4 }} variant="headlineSmall">
            Display Variants
          </Text>
          <View style={{ gap: 8 }}>
            <Text variant="displayLarge">Display Large Text</Text>
            <Text variant="displayMedium">Display Medium Text</Text>
            <Text variant="displaySmall">Display Small Text</Text>
          </View>
        </View>

        <View style={{ marginBottom: 8 }}>
          <Text style={{ marginBottom: 4 }} variant="headlineSmall">
            Headline Variants
          </Text>
          <View style={{ gap: 8 }}>
            <Text variant="headlineLarge">Headline Large Text</Text>
            <Text variant="headlineMedium">Headline Medium Text</Text>
            <Text variant="headlineSmall">Headline Small Text</Text>
          </View>
        </View>

        <View style={{ marginBottom: 8 }}>
          <Text style={{ marginBottom: 4 }} variant="headlineSmall">
            Title Variants
          </Text>
          <View style={{ gap: 8 }}>
            <Text variant="titleLarge">Title Large Text</Text>
            <Text variant="titleMedium">Title Medium Text</Text>
            <Text variant="titleSmall">Title Small Text</Text>
          </View>
        </View>

        <View style={{ marginBottom: 8 }}>
          <Text style={{ marginBottom: 4 }} variant="headlineSmall">
            Label Variants
          </Text>
          <View style={{ gap: 8 }}>
            <Text variant="labelLarge">Label Large Text</Text>
            <Text variant="labelMedium">Label Medium Text</Text>
            <Text variant="labelSmall">Label Small Text</Text>
          </View>
        </View>

        <View style={{ marginBottom: 8 }}>
          <Text style={{ marginBottom: 4 }} variant="headlineSmall">
            Body Variants
          </Text>
          <View style={{ gap: 8 }}>
            <Text variant="bodyLarge">
              Body Large Text - This is a longer text to demonstrate the body text variant styling.
            </Text>
            <Text variant="bodyMedium">
              Body Medium Text - This is a longer text to demonstrate the body text variant styling.
            </Text>
            <Text variant="bodySmall">
              Body Small Text - This is a longer text to demonstrate the body text variant styling.
            </Text>
          </View>
        </View>

        <View style={{ marginBottom: 8 }}>
          <Text style={{ marginBottom: 4 }} variant="headlineSmall">
            Custom Style
          </Text>
          <Text style={{ color: '#e91e63', textAlign: 'center' }} variant="titleLarge">
            Custom Styled Text
          </Text>
        </View>
      </View>
    </ScrollView>
  ),
};
