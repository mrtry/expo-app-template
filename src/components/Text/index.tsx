import React, { FC } from 'react';
import { Text as RNPText, TextProps as RNPTextProps } from 'react-native-paper';

type TextProps = Omit<RNPTextProps<string>, 'variant'> & {
  variant: NonNullable<RNPTextProps<string>['variant']>;
};

export const Text: FC<TextProps> = (props) => {
  return <RNPText {...props} />;
};
