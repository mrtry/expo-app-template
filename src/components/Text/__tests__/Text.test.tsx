import { composeStories } from '@storybook/react';
import { render } from '@testing-library/react-native';
import React from 'react';

// 公式ガイドに従ってcomposeStoriesを使用
import * as stories from '@/components/Text/Text.stories';

// 全てのストーリーをテスト可能なコンポーネントに変換
const { AllVariants } = composeStories(stories);

describe('Text Component', () => {
  it('snapshot', () => {
    const { toJSON } = render(<AllVariants />);
    expect(toJSON()).toMatchSnapshot();
  });
});
