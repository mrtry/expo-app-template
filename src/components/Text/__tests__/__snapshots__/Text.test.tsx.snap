// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Text Component snapshot 1`] = `
<RCTScrollView
  style={
    {
      "flex": 1,
    }
  }
>
  <View>
    <View
      style={
        {
          "gap": 16,
          "padding": 16,
        }
      }
    >
      <View
        style={
          {
            "marginBottom": 8,
          }
        }
      >
        <Text
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              [
                {
                  "fontFamily": "System",
                  "fontSize": 24,
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 32,
                },
                {
                  "marginBottom": 4,
                },
              ],
            ]
          }
        >
          Display Variants
        </Text>
        <View
          style={
            {
              "gap": 8,
            }
          }
        >
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 57,
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 64,
                  },
                  undefined,
                ],
              ]
            }
          >
            Display Large Text
          </Text>
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 45,
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 52,
                  },
                  undefined,
                ],
              ]
            }
          >
            Display Medium Text
          </Text>
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 36,
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 44,
                  },
                  undefined,
                ],
              ]
            }
          >
            Display Small Text
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "marginBottom": 8,
          }
        }
      >
        <Text
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              [
                {
                  "fontFamily": "System",
                  "fontSize": 24,
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 32,
                },
                {
                  "marginBottom": 4,
                },
              ],
            ]
          }
        >
          Headline Variants
        </Text>
        <View
          style={
            {
              "gap": 8,
            }
          }
        >
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 32,
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 40,
                  },
                  undefined,
                ],
              ]
            }
          >
            Headline Large Text
          </Text>
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 28,
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 36,
                  },
                  undefined,
                ],
              ]
            }
          >
            Headline Medium Text
          </Text>
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 24,
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 32,
                  },
                  undefined,
                ],
              ]
            }
          >
            Headline Small Text
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "marginBottom": 8,
          }
        }
      >
        <Text
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              [
                {
                  "fontFamily": "System",
                  "fontSize": 24,
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 32,
                },
                {
                  "marginBottom": 4,
                },
              ],
            ]
          }
        >
          Title Variants
        </Text>
        <View
          style={
            {
              "gap": 8,
            }
          }
        >
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 22,
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 28,
                  },
                  undefined,
                ],
              ]
            }
          >
            Title Large Text
          </Text>
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "500",
                    "letterSpacing": 0.15,
                    "lineHeight": 24,
                  },
                  undefined,
                ],
              ]
            }
          >
            Title Medium Text
          </Text>
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                  undefined,
                ],
              ]
            }
          >
            Title Small Text
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "marginBottom": 8,
          }
        }
      >
        <Text
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              [
                {
                  "fontFamily": "System",
                  "fontSize": 24,
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 32,
                },
                {
                  "marginBottom": 4,
                },
              ],
            ]
          }
        >
          Label Variants
        </Text>
        <View
          style={
            {
              "gap": 8,
            }
          }
        >
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                  undefined,
                ],
              ]
            }
          >
            Label Large Text
          </Text>
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 12,
                    "fontWeight": "500",
                    "letterSpacing": 0.5,
                    "lineHeight": 16,
                  },
                  undefined,
                ],
              ]
            }
          >
            Label Medium Text
          </Text>
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 11,
                    "fontWeight": "500",
                    "letterSpacing": 0.5,
                    "lineHeight": 16,
                  },
                  undefined,
                ],
              ]
            }
          >
            Label Small Text
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "marginBottom": 8,
          }
        }
      >
        <Text
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              [
                {
                  "fontFamily": "System",
                  "fontSize": 24,
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 32,
                },
                {
                  "marginBottom": 4,
                },
              ],
            ]
          }
        >
          Body Variants
        </Text>
        <View
          style={
            {
              "gap": 8,
            }
          }
        >
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "letterSpacing": 0.15,
                    "lineHeight": 24,
                  },
                  undefined,
                ],
              ]
            }
          >
            Body Large Text - This is a longer text to demonstrate the body text variant styling.
          </Text>
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "400",
                    "letterSpacing": 0.25,
                    "lineHeight": 20,
                  },
                  undefined,
                ],
              ]
            }
          >
            Body Medium Text - This is a longer text to demonstrate the body text variant styling.
          </Text>
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 12,
                    "fontWeight": "400",
                    "letterSpacing": 0.4,
                    "lineHeight": 16,
                  },
                  undefined,
                ],
              ]
            }
          >
            Body Small Text - This is a longer text to demonstrate the body text variant styling.
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "marginBottom": 8,
          }
        }
      >
        <Text
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              [
                {
                  "fontFamily": "System",
                  "fontSize": 24,
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 32,
                },
                {
                  "marginBottom": 4,
                },
              ],
            ]
          }
        >
          Custom Style
        </Text>
        <Text
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              [
                {
                  "fontFamily": "System",
                  "fontSize": 22,
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 28,
                },
                {
                  "color": "#e91e63",
                  "textAlign": "center",
                },
              ],
            ]
          }
        >
          Custom Styled Text
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
